rfcs/text/0000-create-file-elf.md

```diff
@@ -0,0 +1,160 @@
+# 0000: Create the ELF sub-field of the File fieldset
+
+- Stage: **1 (proposal)**
+- Date: **TBD**
+
+Create the Executable Linkable Format (ELF) sub-field, of the `file` top-level fieldset. This document metadata can be used for malware research, as well as coding and other application development efforts.
+
+## Fields
+
+**Stage 0**
+
+This RFC is to create the ELF sub-field within the `file.` fieldset. This will include 25 sub-fields.
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| elf.creation_date | date | Extracted when possible from the file's metadata. Indicates when it was built or compiled. It can also be faked by malware creators. |
+| elf.exports | flattened | List of exported element names and types. |
+| elf.exports.name | keyword | Name of exported symbol |
+| elf.exports.type | keyword | Type of exported symbol |
+| elf.segments | nested | ELF object segment list. |
+| elf.segments.type | keyword | ELF object segment type. |
+| elf.segments.sections | keyword | ELF object segment sections. |
+| elf.header | group | Header information of the ELF file. |
+| elf.header.class | keyword | Header class of the ELF file. |
+| elf.header.data | keyword | Data table of the ELF header. |
+| elf.header.machine | keyword | Machine architecture of the ELF header. |
+| elf.header.os_abi | keyword | NEED TO ADD |
+| elf.header.type | keyword | Header type of the ELF file. |
+| elf.header.version | keyword | Version of the ELF header. |
+| elf.header.abi_version | keyword | Version of the ELF Application Binary Interface (ABI). |
+| elf.header.entrypoint | long | Header entrypoint of the ELF file. |
+| elf.header.object_version | keyword | "0x1" for original ELF files. |
+| elf.imports | flattened | List of imported element names and types. |
+| elf.imports.name | keyword | Name of imported symbol |
```

Comment:
Type `flattened` won't allow explicit field mappings to be defined, so I don't think it's necessary to explicitly list them here.

However, is there intent to still describe for data sources on how to "shape" the data for these flattened fields? There are no `type: flattened` fields today in ECS, so how to best capture provide that type of guidance will need to be hashed out.

I don't feel like it needs to be a blocker here, but can we capture this as an outstanding item in the `Concerns` section to discuss further?

Reply:
Yep, got it.